src/tir/transforms/hoist_if_then_else.cc

```diff
@@ -147,26 +197,23 @@ class HoistCandidateSelector final : public StmtExprVisitor {
 
     if (CheckValidIf()) {
       // Check corresponding for loop
-      bool match_found = false;
-      size_t match_for_loop_pos = 0;
+      int match_for_loop_pos = -1;
       for (auto var : if_var_list_) {
-        for (size_t i = 0; i < ordered_for_list_.size() - 1; ++i) {
-          if (ordered_for_list_[i] == var_for_map_[var]) {
+        for (int i = 0; i < static_cast<int>(ordered_list_.size()); ++i) {
```

Comment:
Suggestion: A better way to write this kind of loops is like `for (int i = 0, i_end = static_cast<int>(xxx.size()); i < i_end; i++)`. This is because it is hard to optimize for loop invariant in a C++ compiler. Usually `.size()` will be called in every iteration. This is just a suggestion. You can keep your original code since this is not a performance bottleneck.

Reply:
Thanks for the suggestion! I totally agree with you. However as it is not a bottleneck as you mentioned. Let us keep it.